ament_cpplint/ament_cpplint/main.py

```diff
@@ -144,7 +148,7 @@ def main(argv=sys.argv[1:]):
         else:
             print("Not using '--root'")
         print('')
-        arguments += files
+        arguments += [f for f in files if any(excl not in f for excl in args.exclude)]
```

Comment:
As I have explained above with an example `in` / substring is not a sufficient check for what you want to do. In order to match exact files as well as directories as described in the help string you need to check for `f in args.exclude` or if any of the excludes matches a parent path of `f`.